src/paho/mqtt/client.py

```diff
@@ -625,6 +624,196 @@ def __init__(self, client_id="", clean_session=None, userdata=None,
     def __del__(self):
         self._reset_sockets()
 
+    @property
+    def host(self) -> str:
+        "Host we try to connect to. If don't yet called connect() return empty string"
+        return self._host
+
+    @host.setter
+    def host(self, value: str) -> None:
+        """
+        Update host. This will only be used on future (re)connection. You should probably
+        use reconnect() to update the connection if established.
+        """
+        if value is None or len(value) == 0:
+            raise ValueError("Invalid host.")
+        self._host = value
+
+    @property
+    def port(self) -> int:
+        "TCP port we try to connect to."
+        return self._port
+
+    @port.setter
+    def port(self, value: int) -> None:
+        """
+        Update post. This will only be used on future (re)connection. You should probably
+        use reconnect() to update the connection if established.
+        """
+        if value <= 0:
+            raise ValueError("Invalid port number.")
+        self._port = value
+
+    @property
+    def keepalive(self) -> int:
+        "Keepalive in seconds used by the client "
+        return self._keepalive
+
+    @keepalive.setter
+    def keepalive(self, value: int) -> None:
+        "Update keepalive. It's behavior is undefined if the connection is already open"
+        if self._sock is not None:
+            # The issue here is that the previous value of keepalive matter to possibly
+            # sent ping packet.
+            warnings.warn(
+                "updating keepalive on established connection is not supported",
+                stacklevel=2,
+            )
+            self._sock.settimeout(value)
+
+        if value < 0:
+            raise ValueError("Keepalive must be >=0.")
+
+        self._keepalive = value
+
+    @property
+    def transport(self) -> str:
+        'Transport used for the connection, could be "tcp" or  "websockets".'
```

Comment:
I'd rather skip the "could be" since we don't really mandate it.
```suggestion
        """Transport method used for the connection."""
```